lib/src/lib.rs

```diff
@@ -118,22 +120,29 @@ impl<T: CkTransport> TapSigner<T> {
         new_response
     }
 
-    pub fn derive(&mut self, path: Vec<usize>, cvc: String) -> Result<DeriveResponse, Error> {
+    pub fn derive(&mut self, path: Vec<u32>, cvc: String) -> Result<DeriveResponse, Error> {
+        // set most significant bit to 1 to represent hardened path steps
+        let path = path.iter().map(|p| p ^ (1 << 31)).collect();
+        let app_nonce = rand_nonce();
         let (_, epubkey, xcvc) = self.calc_ekeys_xcvc(cvc, &DeriveCommand::name());
-        let cmd = DeriveCommand::for_tapsigner(self.card_nonce.clone(), path, epubkey, xcvc);
-        let resp: Result<DeriveResponse, Error> = self.transport.transmit(cmd);
-        if let Ok(derive_resp) = &resp {
-            self.card_nonce = derive_resp.card_nonce.clone();
-
-            // TODO: verify reponse
-            // The digest:
-            // b'OPENDIME' (8 bytes)
-            // (card_nonce - 16 bytes)
-            // (nonce from command - 16 bytes)
-            // (chain_code - 32 bytes)
-            // must be signed by the master_pubkey
+        let cmd = DeriveCommand::for_tapsigner(app_nonce.clone(), path, epubkey, xcvc);
+        let derive_response: Result<DeriveResponse, Error> = self.transport.transmit(cmd);
+        if let Ok(response) = &derive_response {
+            let card_nonce = self.card_nonce();
+            let sig = &response.sig;
+            let mut message_bytes: Vec<u8> = Vec::new();
+            message_bytes.extend("OPENDIME".as_bytes());
+            message_bytes.extend(card_nonce);
+            message_bytes.extend(app_nonce);
+            message_bytes.extend(&response.chain_code);
+            let message = Message::from_hashed_data::<sha256::Hash>(message_bytes.as_slice());
+            let signature = Signature::from_compact(sig.as_slice())?;
+            let pubkey = PublicKey::from_slice(response.master_pubkey.as_slice())?;
+            // TODO fix verify when a derivation path is used, currently only works if no path given
```

Comment:
Fix verifying the sig when a derivation path is used, currently only works if no path given.